storage/database/src/storable.rs

```diff
@@ -194,6 +197,68 @@ impl<T> Borrow<[T]> for StorableVec<T> {
     }
 }
 
+//---------------------------------------------------------------------------------------------------- StorableVec
+/// A [`Storable`] string.
+///
+/// This is a wrapper around a `Cow<'static, str>`
+/// that can be stored in the database.
+///
+/// # Invariant
+/// [`StorableStr::from_bytes`] will panic
+/// if the bytes are not UTF-8. This should normally
+/// not be possible in database operations, although technically
+/// you can call this function yourself and input bad data.
+///
+/// # Example
+/// ```rust
+/// # use cuprate_database::*;
+/// # use std::borrow::Cow;
+/// let string: StorableStr = StorableStr(Cow::Borrowed("a"));
+///
+/// // Into bytes.
+/// let into = Storable::as_bytes(&string);
+/// assert_eq!(into, &[97]);
+///
+/// // From bytes.
+/// let from: StorableStr = Storable::from_bytes(&into);
+/// assert_eq!(from, string);
+/// ```
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash, bytemuck::TransparentWrapper)]
+#[repr(transparent)]
+pub struct StorableStr(pub Cow<'static, str>);
```

Comment:
This struct was also added.

Basically the same as `StorableVec` but for strings. `Cow<'static, str>` instead of `String` to allow `Cow::Borrowed("static string")`.

Reason why this was added was because I wondering if we should do custom string comparisons, but chose not to. Can remove this.

Reply:
Probably alright to stay.